src/shared/sol-conffile.c

```diff
@@ -917,9 +943,162 @@ _load_vector_defaults(void)
 }
 
 static int
+iterate_dir_cb(void *data, const char *dir_path, struct dirent *en)
+{
+    char path[PATH_MAX], *sep, *endptr = NULL, *str;
+    struct sol_ptr_vector aliases = SOL_PTR_VECTOR_INIT;
+    struct sol_buffer *file_contents = data;
+    struct sol_json_scanner scanner;
+    struct sol_json_token value;
+    enum sol_json_loop_reason end_reason;
+    const size_t extension = strlen(".json");
```

Comment:
On 22-04-2016 13:48, Flávio Vinicius Alvares Ceolin wrote:
> In src/shared/sol-conffile.c
> <https://github.com/solettaproject/soletta/pull/1919#discussion_r60768904>:
>
>> @@ -917,9 +943,162 @@ _load_vector_defaults(void)
>>  }
>>
>>  static int
>> +iterate_dir_cb(void *data, const char *dir_path, struct dirent *en)
>> +{
>> +    char path[PATH_MAX], *sep, *endptr = NULL, *str;
>> +    struct sol_ptr_vector aliases = SOL_PTR_VECTOR_INIT;
>> +    struct sol_buffer *file_contents = data;
>> +    struct sol_json_scanner scanner;
>> +    struct sol_json_token value;
>> +    enum sol_json_loop_reason end_reason;
>> +    const size_t extension = strlen(".json");
>
> Why don't you use |sizeof()| ? |sizeof| count the size in compile time
> while |strlen| is in run time. Maybe the compilers are smart enough to
> do this in runtime, but I think it's better do not presume it.

But remember `sizeof` will include `\0` on its counting.

>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/solettaproject/soletta/pull/1919/files/04af90d1fd9a253f977ccf193fe3069b17b42406#r60768904>
>